pkg/statsd/handler_http_forwarder_v2.go

```diff
@@ -217,14 +227,9 @@ func (hfh *HttpForwarderHandlerV2) emitMetrics(statser stats.Statser) {
 
 func (hfh *HttpForwarderHandlerV2) Run(ctx context.Context) {
 	var wg wait.Group
-	defer wg.Wait()
 	wg.StartWithContext(ctx, hfh.consolidator.Run)
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case metricMaps := <-hfh.consolidatedMetrics:
+	wg.Start(func() {
+		for metricMaps := range hfh.consolidatedMetrics {
 			mergedMetricMap := mergeMaps(metricMaps)
 			mms := mergedMetricMap.SplitByTags(hfh.dynHeaderNames)
 			for dynHeaderTags, mm := range mms {
```

Comment:
If `ctx` is cancelled, `acquireSem` will fail to get the sem, which means anything pending in the `mm`, `mms`, and `hfh.consolidatedMetrics` won't be processed.

Same with the `hfh.postMetrics` context.

I think it could be handled by removing `ctx` from `acquireSem` and `postMetrics`, then after the `range` has completed, acquiring the semaphore until you own everything (you can probably do a `for len(sem) > 0 { acquire }`.  That will ensure that:
- the `range` has finished, so nothing new will go on
- everything that had a semaphore has released it, so there's nothing pending.

Reply:
- [ ] don't depend on canceled ctx